app/lib/fisbot/code_location_api.rb

```diff
@@ -0,0 +1,13 @@
+class CodeLocationApi < FisbotApi
+  class << self
+    def resource
+      :code_locations
+    end
+
+    def id_from_url(url, branch)
+      query_params = { url: url, branch: branch }
+      uri = api_access.resource_uri('id_from_url', query_params)
```

Comment:
This isn't needed.  The existing eFISbot Code Location API provides querying for the `CodeLocation` with the URL and Branch and the returned data package includes the CodeLocation ID.

See: https://confluence.dc1.lan/display/O2/2016-2017+eFISbot+API+Specification+and+Design#id-2016-2017eFISbotAPISpecificationandDesign-GetsingleCodeLocation